--- a/klcc/klcc.in
+++ b/klcc/klcc.in
@@ -1,6 +1,7 @@
 # -*- perl -*-

 use IPC::Open3;
+use Env qw(KLCC_VERBOSE);

 # Standard includes
 @includes = ("-I${prefix}/${KCROSS}include/arch/${ARCHDIR}",
@@ -112,7 +113,7 @@ $operation = '';		# Compile and link
 undef $lang;

 $save_temps = 0;		# The -save-temps option
-$verbose = 0;			# The -v option
+$verbose = $KLCC_VERBOSE or 0;	# The -v option or KLCC_VERBOSE environment variable
 $shared = 0;	   		# Are we compiling shared?
 $debugging = 0;	   		# -g or -p option present?
 $strip = 0;			# -s option present?
@@ -136,7 +137,7 @@ while ( defined($a = shift(@ARGV)) ) {
	} else {
	    die "$0: unknown option: $a\n";
	}
-    } elsif ( $a =~ /^(-print|-dump|--help|--version|-v)/ ) {
+    } elsif ( $a =~ /^(-print|-dump|--help|--version)/ ) {
	# These share prefixes with some other options, so put this test early!
	# Pseudo-operations; just pass to gcc and don't do anything else
	push(@ccopt, $a);
